libyul/AsmScope.h

```diff
@@ -32,16 +32,12 @@
 #include <functional>
 #include <memory>
 
-namespace dev
-{
-namespace solidity
-{
-namespace assembly
+namespace yul
 {
 
 struct Scope
 {
-	using YulType = yul::YulString;
+	using YulType = YulString;
```

Comment:
Could ``YulString`` be used directly?

Reply:
That's not part of this PR. I'm just attempting to isolate libyul to reduce circular dependencies and making it its own real static library. 

However, I do believe that YulType is still okay, because this is what it is about. That it (however) is represented as a string, is a detail that was abstracted away. :)